src/wms/src/main/java/org/geoserver/wms/legendgraphic/GetLegendGraphicKvpReader.java

```diff
@@ -248,6 +246,12 @@ private LegendRequest addLayer(LayerInfo layerInfo, GetLegendGraphicRequest requ
             LegendInfo legendInfo = resolveLegendInfo(layerInfo.getLegend(),request);
             if(legendInfo != null ){
```

Comment:
This block of code is now repeated 3 times in different places, could it be refactored into a method call?